compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -1148,7 +1148,7 @@ class TypeComparer(initctx: Context) extends ConstraintHandling[AbsentContext] {
       def matchAbstractTypeMember(info1: Type) = info1 match {
         case TypeBounds(lo, hi) if lo ne hi =>
           tp2.refinedInfo match {
-            case rinfo2: TypeBounds =>
+            case rinfo2: TypeBounds if tp1.widenExpr.isSingleton =>
               val ref1 = tp1.widenExpr.select(name)
```

Comment:
To be really correct, shouldn't we check if `ref1` is realizable too ?

Reply:
Yes, we should.